pkg/reconciler/reconciler.go

```diff
@@ -950,8 +961,8 @@ func (r *Reconciler) setupWatches(mgr ctrl.Manager, c controller.Controller) err
 		return err
 	}
 
-	if !r.skipDependentWatches {
-		r.postHooks = append([]hook.PostHook{internalhook.NewDependentResourceWatcher(c, mgr.GetRESTMapper())}, r.postHooks...)
+	if !r.skipDependentWatches || len(r.extraDependentWatches) > 0 {
+		r.postHooks = append([]hook.PostHook{internalhook.NewDependentResourceWatcher(c, mgr.GetRESTMapper(), !r.skipDependentWatches, r.extraDependentWatches...)}, r.postHooks...)
```

Comment:
The renaming of the `NewDependetWatches...` funcs also removes the need for inverted boolean parameters:

```suggestion
       if !r.skipDependentWatches {
		r.postHooks = append([]hook.PostHook{internalhook. NewDependentReleaseResourceWatcher(c, mgr.GetRESTMapper())}, r.postHooks...)
	}

	if len(r.extraDependentWatches) > 0 {
		r.postHooks = append([]hook.PostHook{internalhook. NewDependentResourceWatcherForResources(c, mgr.GetRESTMapper(), r.extraDependentWatches...)}, r.postHooks...)
	}
```